week02/242. Valid Anagram.cpp

```diff
@@ -0,0 +1,17 @@
+//Time complexity -0(n
+//space complexity- 0(s+t)
+class Solution {
+public:
+    bool isAnagram(string s, string t) {
+        unordered_map<int, int> str1;
+        unordered_map<int, int> str2;
+        for (auto &n: s) {
+            str1[n]++;
+        }
+        for (auto &n: t) {
+           str2[n]++;
+        }
+        return str1 == str2;
```

Comment:
```suggestion
        if(s.size() != t.size()) return false;
        vector<int> freq(256, 0);
        for (int i = 0; i < s.size(); i++) {
            freq[s[i]]++;
            freq[t[i]]--;
        }
        for (int i = 0; i < 256; i++) {
            if(freq[i]) return false;
        }
        return true;
```

Very, very costly operations.

1. You do not need `unordered_map`. Just use a vector of size 256 and you're done.
2. The naming of those `unordered_map` looks very odd. Kindly refer to the suggested solution and correct yours.